liboptv/src/orientation.c

```diff
@@ -141,20 +149,803 @@ double weighted_dumbbell_precision(vec2d** targets, int num_targs, int num_cams,
+        ata ((double *) Xh, (double *) XPX, n_obs, numbers, NPAR );
+        
+//         for(i=0;i<NPAR; i++){
+//             for(j=0;j<NPAR;j++){
+//                 printf("XPX[%d][%d] = %lf\n", i,j,XPX[i][j]);
+//             }
+//         }
+        matinv ((double *) XPX, numbers, NPAR);
+        atl ((double *) XPy, (double *) Xh, yh, n_obs, numbers, NPAR);
+        matmul ((double *) beta, (double *) XPX, (double *) XPy, numbers, numbers,1, NPAR,
+                                                                                    NPAR);
+
+        stopflag = 1;
+        for (i=0; i<numbers; i++)
+        {
+         if (fabs (beta[i]) > CONVERGENCE)  stopflag = 0;
+        }
+
+        if ( ! flags->ccflag) beta[6] = 0.0;
+        if ( ! flags->xhflag) beta[7] = 0.0;
+        if ( ! flags->yhflag) beta[8] = 0.0;
+        if ( ! flags->k1flag) beta[9] = 0.0;
+        if ( ! flags->k2flag) beta[10] = 0.0;
+        if ( ! flags->k3flag) beta[11] = 0.0;
+        if ( ! flags->p1flag) beta[12] = 0.0;
+        if ( ! flags->p2flag) beta[13] = 0.0;
+        if ( ! flags->scxflag)beta[14] = 0.0;
+        if ( ! flags->sheflag) beta[15] = 0.0;
+
+        cal->ext_par.x0 += beta[0];
+        cal->ext_par.y0 += beta[1];
+        cal->ext_par.z0 += beta[2];
+        cal->ext_par.omega += beta[3];
+        cal->ext_par.phi += beta[4];
+        cal->ext_par.kappa += beta[5];
+        cal->int_par.cc += beta[6];
+        cal->int_par.xh += beta[7];
+        cal->int_par.yh += beta[8];
+        cal->added_par.k1 += beta[9];
+        cal->added_par.k2 += beta[10];
+        cal->added_par.k3 += beta[11];
+        cal->added_par.p1 += beta[12];
+        cal->added_par.p2 += beta[13];
+        cal->added_par.scx += beta[14];
+        cal->added_par.she += beta[15];
+
+        if(flags->interfflag)
+        {
+          cal->glass_par.vec_x += e1[0]*nGl*beta[16];
+          cal->glass_par.vec_y += e1[1]*nGl*beta[16];
+          cal->glass_par.vec_z += e1[2]*nGl*beta[16];
+          cal->glass_par.vec_x += e2[0]*nGl*beta[17];
+          cal->glass_par.vec_y += e2[1]*nGl*beta[17];
+          cal->glass_par.vec_z += e2[2]*nGl*beta[17];
+        }
+        // beta[0]=beta[0];
+    } // end of while iterations and stopflag
+
+    /* compute residuals etc. */
+
+    matmul ( (double *) Xbeta, (double *) X, (double *) beta, n_obs, numbers, 1, n_obs, 
+                                                                                    NPAR);
+    omega = 0;
+    for (i=0; i<n_obs; i++)
+    {
+        resi[i] = Xbeta[i] - y[i];
+        omega += resi[i] * P[i] * resi[i];
+    }
+    sigma0 = sqrt (omega / (n_obs - numbers));
+
+    for (i=0; i<numbers; i++)  sigmabeta[i] = sigma0 * sqrt(XPX[i][i]);
+
+
+    /* correlations between parameters */
+    /*if (examine)	for (i=0; i<18; i++)
+    {
+      for (j=0; j<18; j++)
+    printf ("%6.2f",
+        XPX[i][j] / (sqrt(XPX[i][i]) * sqrt(XPX[j][j])));
+      printf ("\n");
+    }*/
+
+
+    /* print results */
```

Comment:
Woukd be better to just return the sigmabeta array along with the new calibration. This kind of report is a UI thing, easily done with Python. And killing this part would allow us to get read of the RO define and the use of gradians.